mitmproxy/proxy/server.py

```diff
@@ -427,7 +427,8 @@ def tls_start(tls_start: tls.TlsStartData):
                 tls_start.ssl_conn.set_accept_state()
             else:
                 tls_start.ssl_conn.set_connect_state()
-                tls_start.ssl_conn.set_tlsext_host_name(tls_start.context.client.sni)
+                if tls_start.context.client.sni is not None:
```

Comment:
Do we still support handshakes without SNI present?

Reply:
Yes, we do. The relevant logic is here: https://github.com/mitmproxy/mitmproxy/blob/7d7c2908b2b150e719e3f3c2f086e9290dc0603b/mitmproxy/addons/tlsconfig.py#L99-L106